codalab/worker/bundle_manager.py

```diff
@@ -284,56 +285,56 @@ def _schedule_run_bundles_on_workers(self, workers, user_owned):
             else:
                 workers_list = workers.user_owned_workers(self._model.root_user_id)
 
-            workers_list = self._filter_and_sort_workers(workers_list, bundle)
+            # save _compute_request_X values because somehow they get mutated somewhere down the line (wtf!)
+            # i.e. multiple calls may return different values; which screws up scheduling really badly
+            # TODO: hunt down this bug
```

Comment:
A multithreading issue?  Want to fix before merging?

Reply:
I've been looking into this for 5 hours, narrowing the cause for some wacky scheduling behavior down to this part, and this is where I could no longer make progress. So this is why I decided to implement a work around instead of spending more (unbounded) time to fix this bug.